mava/systems/ippo/networks.py

```diff
@@ -198,34 +201,85 @@ def make_discrete_networks(
         critic_layer_sizes: sizes of hidden layers for the critic network
         observation_network: optional network for processing observations.
             Defaults to utils.batch_concat.
+        activate_mlp_final: Activate final layer of intermediary MLP
+        orthogonal_initialisation: Whether network weights should be
+            initialised orthogonally.
 
     Returns:
         PPONetworks class
     """
 
     num_actions = environment_spec.actions.num_values
 
-    def policy_fn(inputs: jnp.ndarray) -> networks_lib.FeedForwardNetwork:
-        policy_network = hk.Sequential(
-            [
-                observation_network,
-                hk.nets.MLP(policy_layer_sizes, activation=jax.nn.relu),
-                networks_lib.CategoricalHead(
-                    num_values=num_actions, dtype=environment_spec.actions.dtype
-                ),
-            ]
-        )
-        return policy_network(inputs)
-
-    def critic_fn(inputs: jnp.ndarray) -> networks_lib.FeedForwardNetwork:
-        critic_network = hk.Sequential(
-            [
-                observation_network,
-                hk.nets.MLP(critic_layer_sizes, activation=jax.nn.relu),
-                ValueHead(),
-            ]
-        )
-        return critic_network(inputs)
+    if orthogonal_initialisation:
+
+        def policy_fn(inputs: jnp.ndarray) -> networks_lib.FeedForwardNetwork:
+            policy_network = hk.Sequential(
+                [
+                    observation_network,
+                    hk.nets.MLP(
+                        policy_layer_sizes,
+                        activation=jax.nn.relu,
+                        activate_final=activate_mlp_final,
+                        w_init=hk.initializers.Orthogonal(scale=jnp.sqrt(2)),
+                        b_init=hk.initializers.Constant(constant=0.0),
+                    ),
+                    networks_lib.CategoricalHead(
+                        num_values=num_actions,
+                        dtype=environment_spec.actions.dtype,
+                        w_init=hk.initializers.Orthogonal(scale=0.01),
```

Comment:
Does the heads don't need a `b_init`? The code is probably correct. I'm just checking :)

Reply:
This is a good question. From reading the blog post it seems that the heads only get weight initialisation. Also acme's `CategoricalHead` cannot be initialised with a `b-init` parameter which led me to believe it might not be necessary. Please see [here](https://github.com/deepmind/acme/blob/49c8e1b591490d8d36dba21d84b3453cfdb92132/acme/jax/networks/distributional.py#L33).

Reply:
I think we can maybe remove the bias init since the default in haiku for the bias is 0 anyway - [link](https://github.com/deepmind/dm-haiku/blob/307cf7dbda64d637ca423cacc9978f0ca19dc8a6/haiku/_src/basic.py#L156).